lua/entities/gmod_wire_egp/lib/objects/text.lua

```diff
@@ -87,14 +83,27 @@ Obj.Draw = function( self )
 				y = (h * ((self.valign%10)/2))
 			end
 
-			-- Thanks to Wizard for the base to this rotateable text code. I edited it a bit to properly support alignment
+			mat:Identity()
+
+			-- cam.PushModelMatrix replaces the model matrix pushed onto the stack by cam.Start3D2D
+			-- We have to recreate the cam.Start3D2D model matrix
+			if ent:GetClass() == "gmod_wire_egp_emitter" then
```

Comment:
I don't like this hardcoded behaviour. If the draw call needs to know what the base matrix is, then there are two options that I can see:
* the entity should be able to tell it what the matrix is (ie. the code below should only exist in `gmod_wire_egp_emitter`, and be exposed as `ENT:GetEGPDrawMatrix()` or something - other EGP entities would just implement it by returning `Matrix()`.)
* the matrix should be passed in to `Obj:Draw` as a parameter. That way it doesn't have to be calculated by every object that needs it to draw.

Reply:
(If we went the second route, we could then in future pass in a matrix that also applies the object's position and rotation, which would probably simplify quite a lot of object drawing code.)